features/sharding/core/src/test/java/org/apache/shardingsphere/sharding/algorithm/sharding/ShardingAutoTableAlgorithmUtilTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.sharding.algorithm.sharding;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+import org.apache.shardingsphere.infra.datanode.DataNodeInfo;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ShardingAutoTableAlgorithmUtilTest {
+    
+    private Collection<String> collection;
+    
+    private DataNodeInfo dataNodeInfo;
+    
+    @Before
+    public void setup() {
+        collection = new ArrayList<>();
+        collection.add("PREFIX----SUFFIX");
+        collection.add("PREFIXSUFFIXSTRING");
+        collection.add("PREFIX----------");
+        
+        final String prefix = "PREFIX";
```

Comment:
Please remove final modifier in method domain.

Reply:
Fixed this and pushed the changes